specs/Checkbox.md

```diff
@@ -0,0 +1,388 @@
+# Checkbox component specification
+
+The `Checkbox` component allows a user to choose between two mutually exclusive options. 
+
+## Related variant considerations
+
+- Toggle: should be a separate component because it has different anatomy & warrants unique themability.
```

Comment:
GROUP REVIEW:

We should pull the "why" behind stances like this into a Design Principles document.  There is a branch started for this.  @levithomason to update.